src/map/pc.cpp

```diff
@@ -13225,16 +13225,18 @@ short pc_get_itemgroup_bonus(struct map_session_data* sd, t_itemid nameid) {
 	short bonus = 0;
 
 	for (const auto &it : sd->itemgrouphealrate) {
-		uint16 group_id = it.id, i;
-		struct s_item_group_db *group = NULL;
+		uint16 group_id = it.id;
+		std::shared_ptr<s_item_group_db> group = itemdb_group.find(group_id);
 
-		if (!group_id || !(group = itemdb_group_exists(group_id)))
+		if (group_id == 0 || group == nullptr || group->random.empty())
 			continue;
-		
-		for (i = 0; i < group->random[0].data_qty; i++) {
-			if (group->random[0].data[i].nameid == nameid) {
-				bonus += it.val;
-				break;
+
+		if (group->random.count(0) > 0 && !group->random[0]->data.empty()) {
```

Comment:
Is this really intended to only apply to "must" items?
Should we issue a warning on empty here?
I do not think server admins understand this.
Based on documentation I would assume all subgroups...
`bonus2 bAddItemGroupHealRate,ig,n;	Increases HP recovered by n% for items of item group ig`

Reply:
Ha yeah the old random[0] is now random[1] that part is a mistake from me

Reply:
But do you know why we only take the first subgroup into account here?
I think we should loop the subgroups or give better documentation :(

Reply:
I looped all the subgroup in the last commit https://github.com/rathena/rathena/pull/6032/commits/a92c48da8c0e511520eae43800cc9bb8b092a0b5#diff-f0da4401634371005af9888ebaff866f7126cf02aee33b9824dea381a7065e0cR13232
I guess it only checked in 1 subgroup because the groups used for the `bAddItemGroupHealRate` command are only here to group certain items. Theses groups are never used to give a random item, so there is only 1 subgroup for them

Reply:
Nice one :)